src/ucp/api/ucp.h

```diff
@@ -639,6 +682,33 @@ void ucp_context_print_info(ucp_context_h context, FILE *stream);
 ucs_status_t ucp_worker_create(ucp_context_h context, ucs_thread_mode_t thread_mode,
                                ucp_worker_h *worker_p);
 
+/**
+ * @ingroup UCP_WORKER
+ * @brief Create a worker object with list of parameters.
+ *
+ * This routine allocates and initializes a @ref ucp_worker_h "worker" object.
+ * Each worker is associated with one and only one @ref ucp_context_h
+ * "application" context.  In the same time, an application context can create
+ * multiple @ref ucp_worker_h "workers" in order to enable concurrent access to
+ * communication resources. For example, application can allocate a dedicated
+ * worker for each application thread, where every worker can be progressed
+ * independently of others.
+ *
+ * @note The worker object is allocated within context of the calling thread
+ *
+ * @param [in] context     Handle to @ref ucp_context_h
+ *                         "UCP application context".
+ * @param [in] params User defined @ref ucp_worker_params_t parameters
+ *                         for worker creation
+ * @param [out] worker_p   A pointer to the worker object allocated by the
+ *                         UCP library
+ *
+ * @return Error code as defined by @ref ucs_status_t
+ */
+ucs_status_t ucp_worker_create_p(ucp_context_h context,
```

Comment:
old name ucp_worker_create_with_parameters() looked too long. Propose the name please.